presto-hive/src/main/java/com/facebook/presto/hive/metastore/HiveMetastore.java

```diff
@@ -48,4 +54,41 @@
     Optional<Map<String, Partition>> getPartitionsByNames(String databaseName, String tableName, List<String> partitionNames);
 
     Optional<Table> getTable(String databaseName, String tableName);
+
+    Set<String> getRoles(String user);
+
+    Set<HivePrivilege> getDatabasePrivileges(String user, String databaseName);
+
+    Set<HivePrivilege> getTablePrivileges(String user, String databaseName, String tableName);
+
+    default boolean isDatabaseOwner(String user, String databaseName)
+    {
+        // all users are "owners" of the default database
+        if (DEFAULT_DATABASE_NAME.equalsIgnoreCase(databaseName)) {
+            return true;
+        }
+
+        Optional<Database> databaseMetadata = getDatabase(databaseName);
+        if (!databaseMetadata.isPresent()) {
+            return false;
+        }
+
+        Database database = databaseMetadata.get();
+
+        // a database can be owned by a user or role
+        if (database.getOwnerType() == USER && user.equals(database.getOwnerName())) {
```

Comment:
per offline discussion, the ownership should be checked against OWNERSHIP privilege in Hipster.